owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/status/StatusRequestor.kt

```diff
@@ -0,0 +1,113 @@
+
+    companion object {
+        /**
+         * Maximum time to wait for a response from the server when the connection is being tested,
+         * in MILLISECONDs.
+         */
+        private const val TRY_CONNECTION_TIMEOUT: Long = 5000
+        private const val NODE_INSTALLED = "installed"
+        private const val HTTPS_SCHEME = "https"
+        private const val HTTP_SCHEME = "http"
+        private const val NODE_VERSION = "version"
+    }
+
+    private fun checkIfConnectionIsRedirectedToNoneSecure(
+        isConnectionSecure: Boolean,
+        baseUrl: String,
+        redirectedUrl: String
+    ): Boolean {
+        return isConnectionSecure ||
+                (baseUrl.startsWith(HTTPS_SCHEME) && redirectedUrl.startsWith(
+                    HTTP_SCHEME
+                ))
+    }
+
+    fun updateLocationWithRedirectPath(oldLocation: String, redirectedLocation: String): String {
+        if (!redirectedLocation.startsWith("/"))
+            return redirectedLocation
+        val oldLocation = URL(oldLocation)
+        return URL(oldLocation.protocol, oldLocation.host, oldLocation.port, redirectedLocation).toString()
+    }
+
+    private fun getGetMethod(url: String): GetMethod {
+        return GetMethod(URL(url + OwnCloudClient.STATUS_PATH)).apply {
+            setReadTimeout(TRY_CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+            setConnectionTimeout(TRY_CONNECTION_TIMEOUT, TimeUnit.SECONDS)
+        }
+    }
+
+    data class RequestResult(
+        val getMethod: GetMethod,
+        val status: Int,
+        val result: RemoteOperationResult<OwnCloudVersion>,
+        val redirectedToUnsecureLocation: Boolean
+    )
+
+    fun requestAndFollowRedirects(baseLocation: String, client: OwnCloudClient): RequestResult {
+        var currentLocation = baseLocation
+        var redirectedToUnsecureLocation = false
+        var status: Int
+
+        while (true) {
+            val getMethod = getGetMethod(currentLocation)
+
+            status = client.executeHttpMethod(getMethod)
+            val result =
+                if (isSuccess(status)) RemoteOperationResult<OwnCloudVersion>(RemoteOperationResult.ResultCode.OK)
+                else RemoteOperationResult(getMethod)
+
+            if (result.redirectedLocation.isNullOrEmpty() || result.isSuccess) {
+                return RequestResult(getMethod, status, result, redirectedToUnsecureLocation)
+            } else {
+                val nextLocation = updateLocationWithRedirectPath(currentLocation, result.redirectedLocation)
+                redirectedToUnsecureLocation =
+                    checkIfConnectionIsRedirectedToNoneSecure(
+                        redirectedToUnsecureLocation,
+                        currentLocation,
+                        nextLocation
+                    )
+                currentLocation = nextLocation
+            }
+        }
+    }
+
+    private fun isSuccess(status: Int): Boolean = status == HttpConstants.HTTP_OK
```

Comment:
You can shorten it by

    private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK

Reply:
or an extension function, when you use it often, eg

fun Int.isSuccess() = this == HttpConstants.HTTP_OK

Reply:
done

Reply:
I like the extension approach, but since each operation has its own success status, i think it is not needed at all.

> You can shorten it by
> 
> ```
> private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK
> ```

I would go with the first solution suggested instead of a private extension.